### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by Charlotte-Ketteler**

### CODE REVIEW

**Comment by Charlotte-Ketteler** [comment]
Please note that I have changed the handling of the app a bit. Also I created the Branch Tests because I was unsure if I would 
 change (with the tests) to many things and didn't want to lose my already working code. So don't be confused by the Branch name. I will delete the other Branch "CreateAppointment".  Next to the three added tests the edit, create and delete functions are all working. The perspective has changed from user to admin. The admin can book timeslots for the riders and can change the appointment. Other than explained in the US4 there is no "ändern" Button. You just have to click on the already booked appointment to edit it.

File: components/TimeSlotsButton/index.test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    },
+  ],
+};
+
+test("toggles the form when the button is clicked", () => {
+  render(<TimeslotButton timeslot={timeslot} />);
+  const button = screen.getByText("08:00 - 08:50");
+  fireEvent.click(button);
+  expect(button).toBeInTheDocument();
+});
```
Comment (Kheirah) [inline components/TimeSlotsButton/index.test.js:38]: Not quite! 😆 

The test says: _"toggles the form when the button is clicked"._
What your test does: checks if the very button that you click is displayed.

The button is displayed regardless. No matter if you click it or not, it's there. You will need to check what else is displayed after the button is clicked, which is the form. See if you can find something inside the form that you can check to be in the document after that click.

File: pages/BookingPage/[timeslot_id]/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  const ah = [
+    ...booking.horses.map((horse) =>
+      HorsesList.find((h) => h.id === horse.horseId)
+    ),
+    ...HorsesList.filter(
+      (horse) => !horsesAlreadySelected.includes(horse.id)
+    ),
+  ];
+  
+  console.log(ah);
```
Comment (Kheirah) [inline pages/BookingPage/[timeslot_id]/index.js:43]: The `console.log` can be removed 😉

**Review by Kheirah — CHANGES_REQUESTED** [review]
Nice!

But one test needs minor adjustment. See below for more details.

File: components/TimeSlotsButton/index.test.js (hunk 1 above)
Comment (Charlotte-Ketteler) [reply components/TimeSlotsButton/index.test.js:38]: Not sure if I understand but is this what you mean:
Screenshot 2023-09-06 at 11 34 56

Comment (Kheirah) [reply components/TimeSlotsButton/index.test.js:38]: Exactly!

File: pages/BookingPage/[timeslot_id]/index.js (hunk 2 above)
Comment (Charlotte-Ketteler) [reply pages/BookingPage/[timeslot_id]/index.js:43]: Sorry, didn't see that console :D

**Review by Kheirah — APPROVED** [review]
LGTM!

**Pull request merged**